crates/common/src/transaction.rs

```diff
@@ -0,0 +1,658 @@
+use stark_hash::{Felt, HashChain};
+
+use crate::{felt_bytes, prelude::*};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct Transaction {
+    pub hash: TransactionHash,
+    pub variant: TransactionVariant,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum TransactionVariant {
+    DeclareV0(DeclareTransactionV0V1),
+    DeclareV1(DeclareTransactionV0V1),
+    DeclareV2(DeclareTransactionV2),
+    // Regenesis: deploy is a legacy variant and can be removed after regenesis.
+    Deploy(DeployTransaction),
+    DeployAccount(DeployAccountTransaction),
+    InvokeV0(InvokeTransactionV0),
+    InvokeV1(InvokeTransactionV1),
+    L1Handler(L1HandlerTransaction),
+}
+
+impl Transaction {
+    /// Note that this calculates and uses the variants hash.
+    pub fn from_variant(variant: TransactionVariant, chain_id: ChainId) -> Self {
```

Comment:
This method seems to be unused, along with `calculate_hash()` below. Do we need this?

Reply:
No we don't need this -- I'd copied this from my other PR. 

I imagine we probably want the former for tests in the future, and the latter if we start using this type elsewhere though.

Reply:
For execution with blockifier we'd need a _subset_ of these hash calculations: we need to be able to calculate hashes for "broadcasted" transactions -- ie. transactions that are allowed to be submitted for estimation/simulation over RPC.

Also, this implementation seems a bit redundant with what we already have implemented for the gateway transaction type.

Reply:
> Also, this implementation seems a bit redundant with what we already have implemented for the gateway transaction type.

Definitely is -- I just wanted it to be simpler i.e. this drops support for older txns.

Feel free to delete the function; we can always fetch it back from the other PR.

Reply:
I'd leave this out until we actually find a use case for that -- either in the executor or elsewhere...

I'll remove these functions and merge the PR if you're OK with that.

Reply:
Perfect.